k8s/prometheus/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,146 @@
 actions:
-- name: dummy test
+
+- name: Wait 60s to let Prometheus collect metrics
   bashTest:
     script: |-
-      exit 0
+      sleep 60
     expect:
       exitCode:
         equals: 0
+
+- name: Prometheus UI is available
+  bashTest:
+    script: |-
+      prometheus_ui_url="http://${APP_INSTANCE_NAME}-prometheus:9090/graph"
+      curl -s -f "${prometheus_ui_url}"
+    expect:
+      exitCode:
+        equals: 0
+
+- name: Prometheus metrics for its own StatefulSets are available
+  bashTest:
+    script: |-
+      prometheus_api_url="http://${APP_INSTANCE_NAME}-prometheus:9090/api/v1/query"
+      query="kube_statefulset_replicas\{app_kubernetes_io_name=\"${APP_INSTANCE_NAME}\"\}"
+      metrics_found="$(curl -s "${prometheus_api_url}?query=${query}" \
+        | jq '.data.result' | jq length)"
+      if [[ "${metrics_found}" -gt 0 ]]; then
+        echo "OK - Prometheus StatefulSets metrics found"
```

Comment:
Can we just `exit 1` here and modify stdout to expect `ExitCode` equal to 0?

Reply:
It seems that stdout is more verbose, please keep it with stdout.